src/main/java/org/mockito/internal/util/Primitives.java

```diff
@@ -44,7 +44,7 @@ public static boolean isPrimitiveOrWrapper(Class<?> type) {
 
     public static boolean isAssignableFromWrapper(Class<?> valueClass, Class<?> referenceType) {
         if(isPrimitiveOrWrapper(valueClass) && isPrimitiveOrWrapper(referenceType)) {
-            return Primitives.primitiveTypeOf(valueClass).isAssignableFrom(referenceType);
+            return Primitives.primitiveTypeOf(valueClass).isAssignableFrom(Primitives.primitiveTypeOf(referenceType));
```

Comment:
This seemed to be the most sensible fix. It did not break any other test :thinking:

Reply:
If it does not break tests, we're good ;) This strategy applies to 99% of our codebase. We have few weak spots that are not covered with tests (like OSGi bundle configuration), but primitives handling should be good.

Thanks for checking!